plutarch-test/goldens/plutustype.deconstr.bench.golden

```diff
@@ -0,0 +1,26 @@
+matching.typed.newtype {"exBudgetCPU":597830,"exBudgetMemory":1164,"scriptSizeBytes":30}
+matching.typed.sumtype(ignore-fields) {"exBudgetCPU":1659797,"exBudgetMemory":3798,"scriptSizeBytes":44}
+matching.typed.sumtype(partial-match) {"exBudgetCPU":1659797,"exBudgetMemory":3798,"scriptSizeBytes":44}
+matching.typed.sumtype(exhaustive).minting {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":83}
+matching.typed.sumtype(exhaustive).spending {"exBudgetCPU":1749116,"exBudgetMemory":4098,"scriptSizeBytes":93}
+matching.typed.sumtype(exhaustive).rewarding {"exBudgetCPU":2244261,"exBudgetMemory":5300,"scriptSizeBytes":90}
+matching.typed.sumtype(exhaustive).certifying {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":85}
+matching.typed.sumtype(exhaustive)(ignore-fields).minting {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":83}
+matching.typed.sumtype(exhaustive)(ignore-fields).spending {"exBudgetCPU":1749116,"exBudgetMemory":4098,"scriptSizeBytes":93}
+matching.typed.sumtype(exhaustive)(ignore-fields).rewarding {"exBudgetCPU":2244261,"exBudgetMemory":5300,"scriptSizeBytes":90}
+matching.typed.sumtype(exhaustive)(ignore-fields).certifying {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":85}
+matching.raw.newtype {"exBudgetCPU":597830,"exBudgetMemory":1164,"scriptSizeBytes":30}
+matching.raw.sumtype(ignore-fields) {"exBudgetCPU":1122748,"exBudgetMemory":2466,"scriptSizeBytes":33}
+matching.raw.sumtype(partial-match) {"exBudgetCPU":1481159,"exBudgetMemory":3198,"scriptSizeBytes":39}
+matching.raw.sumtype(exhaustive).minting {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":83}
+matching.raw.sumtype(exhaustive).spending {"exBudgetCPU":1749116,"exBudgetMemory":4098,"scriptSizeBytes":93}
+matching.raw.sumtype(exhaustive).rewarding {"exBudgetCPU":2244261,"exBudgetMemory":5300,"scriptSizeBytes":90}
+matching.raw.sumtype(exhaustive).certifying {"exBudgetCPU":2739406,"exBudgetMemory":6502,"scriptSizeBytes":85}
+matching.raw.sumtype(exhaustive)(ignore-fields).minting {"exBudgetCPU":2291676,"exBudgetMemory":5470,"scriptSizeBytes":75}
+matching.raw.sumtype(exhaustive)(ignore-fields).spending {"exBudgetCPU":1301386,"exBudgetMemory":3066,"scriptSizeBytes":85}
+matching.raw.sumtype(exhaustive)(ignore-fields).rewarding {"exBudgetCPU":1796531,"exBudgetMemory":4268,"scriptSizeBytes":82}
+matching.raw.sumtype(exhaustive)(ignore-fields).certifying {"exBudgetCPU":2291676,"exBudgetMemory":5470,"scriptSizeBytes":77}
```

Comment:
@TotallyNotChase Some of these don't match between typed and raw; could you review?

Reply:
Yeah all of these numbers are exactly the same as before - no worries here. We do have some potential inefficiencies in a few places of the `pmatchRepr` machinery that we cannot fix without linear types.